pkg/ds/daemon_set.go

```diff
@@ -204,6 +207,13 @@ func (ds *daemonSet) WatchDesires(
 				err = ds.removePods()
 				if err != nil {
 					err = util.Errorf("Unable to remove pods from intent tree: %v", err)
+					// We must still try to add pods so that we can label the pods that
+					// potentially need to be scheduled because all updates are taken in
+					// only once
```

Comment:
The code has changed to update and remove pods as accordingly when retrying the replication.